### PR TITLE
Implement the ROS2 sink for the TextLogging

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This  PR will allow using the ros2 text logging mechanism within blf framework. Moreover, it also exposes the blf logging capabilities in the python bindings

cc @traversaro

### CODE REVIEW

**Comment by traversaro** [comment]
Cool! I was a bit confused as ROS2 internally uses spdlog as well, but as explained in https://answers.ros.org/question/361736/what-is-the-intention-of-using-spdlog-under-the-hood/ the backend of the logging can be different, so it make sense to have this sink as no spdlog detail of ROS2 is exposed in the public API.

File: src/TextLogging/RosImplementation/include/BipedalLocomotion/TextLogging/RosLogger.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+#include <memory>
+#include <mutex>
+#include <string>
+
+#include <spdlog/common.h>
+#include <spdlog/sinks/base_sink.h>
+
+#include <BipedalLocomotion/TextLogging/Logger.h>
+
+#include <rclcpp/rclcpp.hpp>
```
Comment (traversaro) [inline src/TextLogging/RosImplementation/include/BipedalLocomotion/TextLogging/RosLogger.h:20]:
```suggestion
#include <rclcpp/logging.hpp>
```

Nitpick: using non-catch all headers can reduce compilation times (but there is the possibility that in the future the macro is moved from that header).

Comment (GiulioRomualdi) [reply src/TextLogging/RosImplementation/include/BipedalLocomotion/TextLogging/RosLogger.h]: Done in https://github.com/ami-iit/bipedal-locomotion-framework/pull/587/commits/b6d6f5d8a8093230f66573721bc39b2cadc4c9c7

**Review by S-Dafarra — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
macOS failure is fixed by #597

**Pull request merged**